worlds/ladx/LADXR/generator.py

```diff
@@ -170,123 +176,127 @@ def generateRom(args, settings, ap_settings, auth, seed_name, logic, rnd=None, m
         patches.overworld.patchOverworldTilesets(rom)
         patches.overworld.createDungeonOnlyOverworld(rom)
-    elif settings.overworld == 'nodungeons':
+    elif world.ladxr_settings.overworld == 'nodungeons':
         patches.dungeon.patchNoDungeons(rom)
-    elif settings.overworld == 'random':
+    elif world.ladxr_settings.overworld == 'random':
         patches.overworld.patchOverworldTilesets(rom)
-        mapgen.store_map(rom, logic.world.map)
+        mapgen.store_map(rom, world.ladxr_logic.world.map)
     #if settings.dungeon_items == 'keysy':
     #    patches.dungeon.removeKeyDoors(rom)
     # patches.reduceRNG.slowdownThreeOfAKind(rom)
     patches.reduceRNG.fixHorseHeads(rom)
     patches.bomb.onlyDropBombsWhenHaveBombs(rom)
-    if ap_settings['music_change_condition'] == MusicChangeCondition.option_always:
+    if world.options.music_change_condition == MusicChangeCondition.option_always:
         patches.aesthetics.noSwordMusic(rom)
-    patches.aesthetics.reduceMessageLengths(rom, rnd)
+    patches.aesthetics.reduceMessageLengths(rom, world.random)
     patches.aesthetics.allowColorDungeonSpritesEverywhere(rom)
-    if settings.music == 'random':
-        patches.music.randomizeMusic(rom, rnd)
-    elif settings.music == 'off':
+    if world.ladxr_settings.music == 'random':
+        patches.music.randomizeMusic(rom, world.random)
+    elif world.ladxr_settings.music == 'off':
         patches.music.noMusic(rom)
-    if settings.noflash:
+    if world.ladxr_settings.noflash:
         patches.aesthetics.removeFlashingLights(rom)
-    if settings.hardmode == "oracle":
+    if world.ladxr_settings.hardmode == "oracle":
         patches.hardMode.oracleMode(rom)
-    elif settings.hardmode == "hero":
+    elif world.ladxr_settings.hardmode == "hero":
         patches.hardMode.heroMode(rom)
-    elif settings.hardmode == "ohko":
+    elif world.ladxr_settings.hardmode == "ohko":
         patches.hardMode.oneHitKO(rom)
-    if settings.superweapons:
+    if world.ladxr_settings.superweapons:
         patches.weapons.patchSuperWeapons(rom)
-    if settings.textmode == 'fast':
+    if world.ladxr_settings.textmode == 'fast':
         patches.aesthetics.fastText(rom)
-    if settings.textmode == 'none':
+    if world.ladxr_settings.textmode == 'none':
         patches.aesthetics.fastText(rom)
         patches.aesthetics.noText(rom)
-    if not settings.nagmessages:
+    if not world.ladxr_settings.nagmessages:
         patches.aesthetics.removeNagMessages(rom)
-    if settings.lowhpbeep == 'slow':
+    if world.ladxr_settings.lowhpbeep == 'slow':
         patches.aesthetics.slowLowHPBeep(rom)
-    if settings.lowhpbeep == 'none':
+    if world.ladxr_settings.lowhpbeep == 'none':
         patches.aesthetics.removeLowHPBeep(rom)
-    if 0 <= int(settings.linkspalette):
-        patches.aesthetics.forceLinksPalette(rom, int(settings.linkspalette))
+    if 0 <= int(world.ladxr_settings.linkspalette):

```

Comment:
I believe this is another unneeded cast
```suggestion
    if 0 <= world.ladxr_settings.linkspalette:
```

Reply:
This one is a string, so the cast actually is needed.